content/post/2022-02-21-code-style/index.Rmd

```diff
@@ -0,0 +1,173 @@
+    It defines code smells and refactoring techniques.
+    See also the blogpost ["Explaining Variable" by Pete Hodgson](https://blog.thepete.net/blog/2021/06/24/explaining-variable/) (heard of via [Jenny Bryan](https://twitter.com/JennyBryan/status/1412140590842597385)).
+
+-   [The Programmer's Brain by Felienne Hermans](https://www.goodreads.com/book/show/57196550-the-programmer-s-brain) gives a perspective on e.g. how code smells or bad naming influence cognitive load.
+    It is full of practical tips.
+
+## Tools
+
+To begin with the most important, or basic, tool here is probably to use an IDE, integrated development environment, like RStudio IDE or VSCODE, as IDEs come with, or easily support, code diagnostic tools.
+
+### Diagnostic tools
+
+#### lintr
+
+Linting tools will indicate errors and potentially style preference violations.
+
+The [lintr R package by Jim Hester](https://github.com/r-lib/lintr) has a lot of useful linters such as whether code is commented, whether lines are too long etc.
+
+It is pretty common to find *linting tools* in all editors (VSCODE and others) and for most languages.
+To set up lintr with your code editor refer to [its docs](https://github.com/r-lib/lintr#editors-setup).
+Unless you have RStudio IDE and linting is the one included already in the IDE (for errors mainly only, not style preference).
+If you use VSCODE see <https://github.com/REditorSupport/vscode-R/wiki/R-Language-Service#formatting>.
+
+To use lintr in testthat see [`expect_lint()`](https://rdrr.io/cran/lintr/man/expect_lint.html).
+
+#### pkgcheck
+
+One of the aspects checked by rOpenSci Mark Padgham's [pkgcheck package](https://docs.ropensci.org/pkgcheck/index.html) is *"Left-assign operators must be used consistently throughout all code (so either all = or all \<-, but not a mixture of both)."*.
+
+### Fixing tools
+
+#### RStudio IDE shortcut for code indentation
+
+In RStudio IDE selecting code and hitting `Ctrl + I` will re-indent code for you!
+
+#### styler
+
+The [styler R package](https://styler.r-lib.org) automatically reformats code.
+Its documentation includes a handy vignette on [customizing styler](https://styler.r-lib.org/articles/customizing_styler.html), for when preferences differ from the default.
+Examples:
+
+-   the [tiny spaceout R package](https://github.com/ropensci-review-tools/spaceout) that adds spaces between code references.
+-   the [grkstyle R package](https://github.com/gadenbuie/grkstyle) holding Garrick Aden-Buie's personal preferences.
+-   the [styler.mlr R package](https://github.com/mlr-org/styler.mlr) implementing the mlr style guide.
+
+The styler package documents some [third-part integration](https://styler.r-lib.org/articles/third-party-integrations.html): you don't have to remember to run styler manually.
+In particular, to use styler on its own CI workflow on GitHub Actions: <https://github.com/r-lib/actions/tree/v2-branch/examples#style-package>
+
```

Comment:
I don't know how you would include a not like box (or maybe this should be in footnote) but I would add a disclaimer on: 

With fixing tools, your original source code will be modified. It is really adviced to use version control with your project - but (place emoji here) who does not use version control when coding nowadays ? If you do not, no worries we got you covered (place link to awesome blog post) 

What do you think ?